### PR TITLE
KafkaChannel upgrade tests

### PR DESCRIPTION
**Pull request by aliok**
## Proposed Changes

- Upgrade tests for KafkaChannel
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by aliok** [comment]
```
&{{KafkaChannel messaging.knative.dev/v1beta1} {sut  eventing-e2e11  1bfc08c1-a179-4eed-88d9-e4d281d26d82 12783 1 2022-03-31 11:02:44 +0000 UTC <nil> <nil> map[] map[messaging.knative.dev/subscribable:v1] [] [kafkachannels.messaging.knative.dev]  [{Go-http-client Update messaging.knative.dev/v1beta1 2022-03-31 11:02:44 +0000 UTC FieldsV1 {"f:metadata":{"f:annotations":{".":{},"f:messaging.knative.dev/subscribable":{}}},"f:spec":{".":{},"f:delivery":{".":{},"f:backoffDelay":{},"f:backoffPolicy":{},"f:retry":{}},"f:numPartitions":{},"f:replicationFactor":{},"f:retentionDuration":{}}} } {kafka-controller Update messaging.knative.dev/v1beta1 2022-03-31 11:02:44 +0000 UTC FieldsV1 {"f:metadata":{"f:finalizers":{".":{},"v:\"kafkachannels.messaging.knative.dev\"":{}}}} } {kafka-controller Update messaging.knative.dev/v1beta1 2022-03-31 11:02:44 +0000 UTC FieldsV1 {"f:status":{".":{},"f:conditions":{},"f:observedGeneration":{}}} status}]} {1 [{Addressable Unknown  {2022-03-31 11:02:44 +0000 UTC}  } {ConfigMapUpdated Unknown  {2022-03-31 11:02:44 +0000 UTC}  } {ConfigParsed Unknown  {2022-03-31 11:02:44 +0000 UTC}  } {DataPlaneAvailable False  {2022-03-31 11:02:44 +0000 UTC} Data plane not available Did you install the data plane for this component?} {ProbeSucceeded Unknown  {2022-03-31 11:02:44 +0000 UTC}  } {Ready False  {2022-03-31 11:02:44 +0000 UTC} Data plane not available Did you install the data plane for this component?} {TopicReady Unknown  {2022-03-31 11:02:44 +0000 UTC}  }] map[]}}
```

Interesting...

**Comment by pierDipi** [comment]
@aliok see https://github.com/knative-sandbox/eventing-kafka-broker/blob/1695e55f3f3ab27b6f941cdbef38049874f2142a/test/e2e-common.sh#L134

**Comment by aliok** [comment]
LOL I found it around the same time you wrote me :)

File: test/upgrade/continual.go [hunk 1]
```diff
@@ -38,6 +40,17 @@ func BrokerContinualTests() []pkgupgrade.BackgroundOperation {
 	}
 }
 
+// ChannelContinualTests returns background operations to test KafkaChannel
+// functionality in continual manner during the whole upgrade and downgrade
+// process asserting that all events are propagated.
+func ChannelContinualTests() []pkgupgrade.BackgroundOperation {
+	// don't want to run the Broker-Backed-By-KafkaChannel tests,
+	// so, we don't call the test method directly
```
Comment (pierDipi) [inline test/upgrade/continual.go:48]: why not ? my naive thinking says that it increases coverage.

File: test/upgrade/suite.go [hunk 2]
```diff
@@ -28,14 +28,17 @@ func Suite() pkgupgrade.Suite {
...
 				SinkPreUpgradeTest(),
 			},
 			PostUpgrade: []pkgupgrade.Operation{
 				BrokerPostUpgradeTest(),
+				ChannelPostUpgradeTest(),
 				SinkPostUpgradeTest(),
 			},
 			PostDowngrade: []pkgupgrade.Operation{
 				BrokerPostDowngradeTest(),
+				//ChannelPostDowngradeTest(),
```
Comment (pierDipi) [inline test/upgrade/suite.go:41]: Why is it commented out?
```suggestion
				ChannelPostDowngradeTest(),
```

File: test/upgrade/preupgrade.go [hunk 3]
```diff
@@ -21,15 +21,22 @@ import (
 
 	eventing "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/eventing/v1alpha1"
 	"knative.dev/eventing-kafka-broker/test/e2e"
+
+	eventingkafkaupgrade "knative.dev/eventing-kafka/test/upgrade"
 )
 
-// BrokerPreUpgradeTest tests channel operations before upgrade.
+// BrokerPreUpgradeTest tests broker basic operations before upgrade.
```
Comment (pierDipi) [inline test/upgrade/preupgrade.go:28]: :man_facepalming: :see_no_evil:

File: test/upgrade/suite.go (hunk 2 above)
Comment (aliok) [reply test/upgrade/suite.go]: 🤦

File: test/upgrade/continual.go (hunk 1 above)
Comment (aliok) [reply test/upgrade/continual.go:48]: I think proper way to use a broker, if you have this repository is to use KafkaBroker and not MTBroker

Comment (pierDipi) [reply test/upgrade/continual.go:48]: +1 but what if there are existing users using KafkaChannel based brokers, is this setup still supported?

Comment (aliok) [reply test/upgrade/continual.go:48]: I don't know...

I would prefer @matzew to have a look at that

Comment (aliok) [reply test/upgrade/continual.go:48]: I will enable it know and see if it passes w/o additional work. If it does I will unhold.
If there's work, we should discuss if we want to support this.

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve 
/retest (flake unit test)

**Comment by aliok** [comment]
/unhold

ok, test passing, no need to hold

**Pull request merged**